### PR TITLE
Material UI

### PR DESCRIPTION
**Pull request by lotos09**
какая то помойка в useStyles, буду разгребать с утра

### CODE REVIEW

**Comment by lotos09** [comment]
жду ревью:)

File: .idea/inspectionProfiles/Project_Default.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<component name="InspectionProjectProfileManager">
```
Comment (im-ihrynevytskyi) [inline .idea/inspectionProfiles/Project_Default.xml]: Remove pls all files from .idea directory

File: src/App.css [hunk 2]
```diff
@@ -1,38 +1,16 @@
-.App {
+/*
```
Comment (im-ihrynevytskyi) [inline src/App.css]: Do you need all this commented code?

File: src/components/BreedInfo.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import {useSelector} from "react-redux";
+import { makeStyles } from '@material-ui/core';
+import { Paper } from '@material-ui/core';
+import StarIcon from '@material-ui/icons/Star';
+
+const useStyles = makeStyles((theme) => ({
```
Comment (im-ihrynevytskyi) [inline src/components/BreedInfo.js]: Please move it into another file

File: src/components/BreedInfo.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        },
+    },
+}));
+
+export const BreedInfo = () => {
+    const classes = useStyles();
+//
+    const state = useSelector(store=> store);
+    const clickedBreed = useSelector(store=>store.clickedBreed)
+    const selectedBreedObjFromStore = state.breedsSlice.breeds.filter((obj)=> obj.id === clickedBreed);
```
Comment (im-ihrynevytskyi) [inline src/components/BreedInfo.js]: Redundant all store getting on 42 line, it is possible to just 
`const selectedBreedObjFromStore = useSelector(store => store.breedsSlice.breeds.find((obj)=> obj.id === clickedBreed)`

File: src/components/BreedInfo.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+
+    if (clickedBreed === '') {
+        return <p>choose your fighter</p>
+    } else
```
Comment (im-ihrynevytskyi) [inline src/components/BreedInfo.js]: Redundant usage of else, don't need it here

File: src/components/Gallery.js [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+//app components
+import {chooseBreedAction, loadBreedImages, loadBreeds} from './reducer_and_actions';
+import {BreedInfo} from "./BreedInfo";
+
+//MUI
+import {GridListTileBar, Input, makeStyles, Paper, Typography} from '@material-ui/core';
+import {GridList, GridListTile, ListSubheader, CircularProgress} from '@material-ui/core';
+import {NavBar} from "./NavBar";
+
+const useStylesApp = makeStyles((theme) => ({
```
Comment (im-ihrynevytskyi) [inline src/components/Gallery.js]: Move styles into another file

File: src/components/Gallery.js [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+                    return <Breed key={item.id} breed={item}/>
+                })}
+            </GridList>
+        </div>
+    );
+}
+
+const Breed = (props) => {
+    const classes = useStylesApp();
+    const obj = props.breed;
```
Comment (im-ihrynevytskyi) [inline src/components/Gallery.js]: Could use destructuring e.g:
`const Breed = ({breed}) => {`

File: src/components/Gallery.js [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+            handleClick()
+            window.scroll({
+                top: 42,
+                left: 100,
+                behavior: 'smooth'
+            })
+        }}>
+            <img className={classes.gridImg} alt={obj.name} src={obj.image.url}/>
+            <GridListTileBar
+                style={{
```
Comment (im-ihrynevytskyi) [inline src/components/Gallery.js]: Its better to avoid inline styling

File: src/components/Gallery.js [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+                    cursor: 'pointer'
+                }}
+                title={obj.name}
+                onClick={() => handleClick()}
```
Comment (im-ihrynevytskyi) [inline src/components/Gallery.js]: Why not just simply onClick={handleClick}

File: src/components/Home.js [hunk 10]
```diff
@@ -0,0 +1,62 @@
+import React from "react";
+import {Button, makeStyles, Menu, MenuItem} from "@material-ui/core";
+import {useHistory} from "react-router-dom";
+import logo from '../media/don.jpg';
+
+const useStyles = makeStyles((Theme) => ({
```
Comment (im-ihrynevytskyi) [inline src/components/Home.js]: The same as above, move to another file pls

File: src/components/NavBar.js [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+                <Button onClick={galleryClick}>gallery</Button>
+                <Button onClick={dataGridClick}>dataGrid</Button>
+                <Button onClick={tableClick}>table</Button>
+            </ButtonGroup>
+
+        </div>
+    )
+}
+
+/*
```
Comment (im-ihrynevytskyi) [inline src/components/NavBar.js]: Do you need this commented section

File: src/components/Table.js [hunk 12]
```diff
@@ -0,0 +1,253 @@
...
+    CircularProgress, Divider,
+    Input,
+    TablePagination,
+    TableSortLabel
+} from "@material-ui/core";
+import {clearImagesReducer, loadBreedImages, loadBreeds} from "./reducer_and_actions";
+import {NavBar} from "./NavBar";
+//import {useStyles} from "./MUIStyles";
+
+const useStyles = makeStyles((theme) => ({
```
Comment (im-ihrynevytskyi) [inline src/components/Table.js]: The same

File: src/components/Table.js [hunk 13]
```diff
@@ -0,0 +1,253 @@
...
+//input
+    const [inputState, setInputState] = useState('');
+    const inputHandle = ({target}) => {
+        setInputState(target.value);
+    }
+    const filteredBreeds = breedsState.breeds.filter((item) => {
+        return item.image && item.image.url && item.name.toLowerCase().includes(inputState.toLowerCase())
+    });
+
+    filteredBreeds.sort((a, b) => {
```
Comment (im-ihrynevytskyi) [inline src/components/Table.js]: sort function modifies original array, so try to implement in another way

File: src/components/Table.js [hunk 14]
```diff
@@ -0,0 +1,253 @@
...
+        }
+        if (sortState.orderBy === 'affect' && sortState.order === true) {
+            return b.affection_level - a.affection_level;
+        }
+        if (sortState.orderBy === 'child' && sortState.order === false) {
+            return a.child_friendly - b.child_friendly;
+        }
+        if (sortState.orderBy === 'child' && sortState.order === true) {
+            return b.child_friendly - a.child_friendly;
+        } else return filteredBreeds;
```
Comment (im-ihrynevytskyi) [inline src/components/Table.js]: Redundant else usage, could do 
`return filteredBreeds;`

File: src/components/Table.js [hunk 15]
```diff
@@ -0,0 +1,253 @@
...
+                                                active={sortState.orderBy === 'child'}
+                                >child friendly</TableSortLabel>
+                            </TableCell>
+                        </TableRow>
+                    </TableHead>
+                    <TableBody>
+                        {pagedArr.map((row) => {
+                            return <TableRow key={row.id} className={classes.tableRow}
+                                             selected={location.pathname === `/table/${row.id}`}
+                                             onClick={() => {
```
Comment (im-ihrynevytskyi) [inline src/components/Table.js]: Move this code to function

File: src/components/Table.js [hunk 16]
```diff
@@ -0,0 +1,253 @@
...
+
+        </div>
+
+    );
+}
+
+function Info() {
+    const classes = useStyles();
+    let {breedId} = useParams();
+    const selectedBreed = useSelector(store => store.breedsSlice.breeds.filter((item) => item.id === breedId))[0]
```
Comment (im-ihrynevytskyi) [inline src/components/Table.js]: You can use .find array method

File: src/components/Table.js [hunk 17]
```diff
@@ -0,0 +1,253 @@
...
+        return (
+            <div className={classes.infoPage}>
+                <h2>{selectedBreed.name}</h2>
+
+                {Boolean(!images.length) && <div className={classes.spinner}>
+                    <CircularProgress/>
+                </div>}
+                {Boolean(images.length) && <div className={classes.gridList}>
+                    {images.map((item, index) => (
+                        <img className={classes.gridImg} alt='breedImage' src={item} key={index}/>
```
Comment (im-ihrynevytskyi) [inline src/components/Table.js]: Passing index as a key is not good solution

File: src/components/reducer_and_actions.js [hunk 18]
```diff
@@ -0,0 +1,67 @@
+import {getBreedImages} from "./api";
+import {getBreeds} from './api';
+
+
+//breed images
+const initialState = [];
+export const imagesReducer = (state = initialState, action) => {
```
Comment (im-ihrynevytskyi) [inline src/components/reducer_and_actions.js]: Separate reducers and actions into their own files, for example ./src/store/reducers/reducerName.js, ./src/store/actions/reducerActionName.js

File: src/components/theme.js [hunk 19]
```diff
@@ -0,0 +1,27 @@
+const {createMuiTheme} = require("@material-ui/core");
+
+
+
+export const theme = createMuiTheme({
+    overrides: {
+
+        // Style sheet name ⚛️
```
Comment (im-ihrynevytskyi) [inline src/components/theme.js]: do you need this comments?

File: src/components/theme.js [hunk 20]
```diff
@@ -0,0 +1,27 @@
+const {createMuiTheme} = require("@material-ui/core");
```
Comment (im-ihrynevytskyi) [inline src/components/theme.js]: Please use import, not require

**Pull request merged**